notification.c

```diff
@@ -286,3 +286,47 @@ void notification_handle_button(struct mako_notification *notif, uint32_t button
 		break;
 	}
 }
+
+/*
+ * Searches through the notifications list and returns the next position at which to insert based on urgency and direction
+ */
+static struct wl_list *get_last_notif_by_urgency(struct wl_list *notifications, enum mako_notification_urgency urgency, int direction) {
+	enum mako_notification_urgency current = urgency;
+
+	if (wl_list_length(notifications) == 0) {
+		return notifications;
+	}
+
+	while (current <= MAKO_NOTIFICATION_URGENCY_HIGH && current >= MAKO_NOTIFICATION_URGENCY_UNKNOWN) {
+		struct mako_notification *notif;
+		wl_list_for_each_reverse(notif, notifications, link) {
+			if (notif->urgency == current) {
+				return &notif->link;
+			}
+		}
+		current += direction;
+	}
+
+	return notifications;
+}
+
+void insert_notification(struct mako_state *state, struct mako_notification *notif) {
+	struct mako_config *config = &state->config;
+	struct wl_list *insert_node;
+
+	//This works but I don't like the way it's done. Can probably use some work.
```

Comment:
Same here. Probably a better way to do this. It works but it's ugly (to me anyway).